bep_webrtc.rst

```diff
@@ -0,0 +1,241 @@
+process.
+
+::
+
+  peer 1                 tracker                  peer 2
+
+    |                       |                       |
+    |                       |                       |
+    | announce start        |                       |
+    |   numwant=x           |                       |
+    |   x SDP offers        |                       |
+    | >-------------------> |                       |
+    |                       |                       |
+    |                       |  SDP offer            |
+    |                       | >-------------------> |
+    |                       |                       |
+    |                       |  SDP answer           |
+    |                       | <-------------------< |
+    |                       |                       |
+    |  SDP answer           |                       |
+    | <-------------------< |                       |
+    |                       |                       | connection established
+   .|.. .. .. .. .. .. .. .. .. .. .. .. .. .. .. ..|.. .
+    |                                               |
+    V                                               V
+
+Websocket tracker protocol
+==========================
+
+credits
+-------
+The websocket tracker protocol has been designed for the Webtorrent [3] project
+by Feross Aboukhadijeh and hundreds of open source contributors.
+
+overview
+--------
+The websocket tracker uses JSON payloads reflecting the HTTP request parameters
+and an additional action property used to switch between "announce" and other
+actions (ex. scrape). If the announce URL of the torrent contains the ws or wss
+protocol, a websocket connection is established between the client and the
+tracker.
+
+WebRTC offers and answers can be provided as an extension of the announce
+message, the tracker will be responsible to forward them between the peers to
+act as a signaling service.
+
+After the connection is open, the client can begin announcing itself by using
+the JSON messages below.
+
+signaling related message format
+---------------------------------
+announce request::
+
+  {
+    "action":     "announce",
+    "info_hash":  "",
+    "peer_id":    ""
+    "numwant":    0,
+    "uploaded":   0,
+    "downloaded": 0,
+    "left":       0,
+    "event":      "",
+    "offers":     [
+      {
+        "offer_id": "",
+        "offer": ""
+      },
+      ...
+    ]
+  }
+
+A client can provide SDP offers that the tracker will forward to peers to
+establish a connection.
+"offers" is a JSON array containing a list of WebRTC SDP offers and a generated
+"offer_id" used to match peers and offers.
+
+
+announce response::
+
+  {
+    "action":     "announce",
+    "complete":   0,
+    "incomplete": 0,
+    "interval":   0,
+    "info_hash":  ""
+  }
+
+
+offer or answer message::
+
+  {
+    "action":     "announce",
+    "info_hash":  "",
+    "offer_id":   "",
+    "peer_id":    "",
+    "sdp":      ""
```

Comment:
I changed from answer to sdp as this property can be either an offer or a response.